### PR TITLE
fix: paginator works with loop:false

### PR DESCRIPTION
**Pull request by adiktofsugar**
Follow up to https://github.com/SBoudrias/Inquirer.js/pull/925

I'm not sure if I should have changed the Paginator in packages/core. It seems like a copy, and it seems like this repo is moving toward a monorepo style, but I think the one in inquirer/utils that I've changed will fix the actual problem so I'll leave it to the maintainers to copy these changes in, unless you want me to do it.

### CODE REVIEW

**Comment by adiktofsugar** [comment]
This is what it looks like. The first prompt is with loop:false. The second is without.
inquirer-fixed

**Comment by sqren** [comment]
Beautiful!

File: packages/inquirer/lib/utils/paginator.js [hunk 1]
```diff
@@ -7,20 +7,18 @@ var _ = {
...
 /**
- * The paginator keeps track of a pointer index in a list and returns
- * a subset of the choices if the list is too long.
+ * The paginator returns a subset of the choices if the list is too long.
  */
 
 class Paginator {
-  constructor(screen) {
-    this.pointer = 0;
+  constructor(screen, isInfinite) {
```
Comment (SBoudrias) [inline packages/inquirer/lib/utils/paginator.js:14]: Let's pass this as option `{ isInfinite }`. Just a boolean argument won't convey the meaning.

File: packages/inquirer/lib/utils/paginator.js [hunk 2]
```diff
@@ -33,7 +31,21 @@ class Paginator {
     if (lines.length <= pageSize) {
       return output;
     }
+    var getLinesFn = this.isInfinite ? this.getInfiniteLines : this.getFiniteLines;
+    var visibleLines = getLinesFn.call(this, lines, active, pageSize);
```
Comment (SBoudrias) [inline packages/inquirer/lib/utils/paginator.js:35]:
```js
var visibleLines = this.isInfinite
  ? this.getInfiniteLines(lines, active, pageSize)
  : this.getFiniteLines(lines, active, pageSize);
```

File: packages/inquirer/lib/utils/paginator.js (hunk 1 above)
Comment (SBoudrias) [reply packages/inquirer/lib/utils/paginator.js:14]: Also, I believe all major versions on node supports defaults parameter `{ isInfinite = true }` and we can remove the type check default later in the code.

**Comment by SBoudrias** [comment]
Thanks! 🙏

**Pull request merged**